src/OpenIddict.Core/OpenIddictProvider.cs

```diff
@@ -234,6 +240,55 @@ public class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServerProvid
             context.HandleResponse();
         }
 
+        public override async Task ProfileEndpoint([NotNull] ProfileEndpointContext context) {
+            var manager = context.HttpContext.RequestServices.GetRequiredService<OpenIddictManager<TUser, TApplication>>();
+
+            var principal = context.AuthenticationTicket?.Principal;
+            Debug.Assert(principal != null);
+
+            // Note: user may be null if the user has been removed. In this case, return a 400 response.
+            var user = await manager.FindByIdAsync(principal.GetUserId());
+            if (user == null) {
+                context.Response.StatusCode = 400;
+                context.HandleResponse();
+
+                return;
+            }
+
+            // Note: "sub" is a mandatory claim. See http://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse
+            context.Claims[JwtRegisteredClaimNames.Sub] = await manager.GetUserIdAsync(user);
+
+            // Resolve the username and the email address corresponding to the user.
+            var username = await manager.GetUserNameAsync(user);
+            var email = await manager.GetEmailAsync(user);
+
+            if (principal.HasClaim(OpenIdConnectConstants.Claims.Scope, OpenIdConnectConstants.Scopes.Profile)) {
+                // Only include the preferred_username claim if the username
+                // doesn't correspond to the registered email address.
+                if (!string.Equals(username, email, StringComparison.OrdinalIgnoreCase)) {
+                    context.Claims[OpenIddictConstants.Claims.PreferredUsername] = username;
+                }
+
+                context.Claims[JwtRegisteredClaimNames.FamilyName] = await manager.FindClaimAsync(user, ClaimTypes.Surname);
+                context.Claims[JwtRegisteredClaimNames.GivenName] = await manager.FindClaimAsync(user, ClaimTypes.GivenName);
+                context.Claims[JwtRegisteredClaimNames.Birthdate] = await manager.FindClaimAsync(user, ClaimTypes.DateOfBirth);
+            }
+
+            if (principal.HasClaim(OpenIdConnectConstants.Claims.Scope, OpenIdConnectConstants.Scopes.Email)) {
+                context.Claims[JwtRegisteredClaimNames.Email] = email;
+                context.Claims[OpenIddictConstants.Claims.EmailVerified] = await manager.IsEmailConfirmedAsync(user) ? "true" : "false";
+            };
+
```

Comment:
I'll probably add native support for `address` and `phone` before merging this PR: http://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims